t/unit/transport/test_redis.py

```diff
@@ -229,42 +444,97 @@ def _get_response_error(self):
 
     def _new_queue(self, queue, **kwargs):
         for pri in self.priority_steps:
-            self.client._new_queue(self._q_for_pri(queue, pri))
+            self.client._new_queue(self._queue_for_priority(queue, pri))
 
     def pipeline(self):
-        return Pipeline(Client())
+        return FakePipelineLite(FakeRedisClientLite())
+
+
+class FakeRedisKombuChannel(kombu_redis.Channel):
+    _fanout_queues = {}
 
+    def __init__(self, *args, **kwargs):
+        super(FakeRedisKombuChannel, self).__init__(*args, **kwargs)
 
-class Transport(redis.Transport):
-    Channel = Channel
+    def _get_client(self):
+        return FakeRedisClient
+
+    def _create_client(self, asynchronous=False):
+        global _fake_redis_client
+        if _fake_redis_client is None:
+            _fake_redis_client = self._get_client()()
+        return _fake_redis_client
+
+    @contextmanager
+    def conn_or_acquire(self, client=None):
+        if client:
+            yield client
+        else:
+            yield self._create_client()
+
+    def _get_pool(self, asynchronous=False):
+        params = self._connparams(asynchronous=asynchronous)
+        self.keyprefix_fanout = self.keyprefix_fanout.format(db=params['db'])
+        return FakeRedisConnectionPool(**params)
+
+    def _get_response_error(self):
+        return ResponseError
+
+    def _new_queue(self, queue, **kwargs):
+        for pri in self.priority_steps:
+            self.client._new_queue(self._queue_for_priority(queue, pri))
+
+    def pipeline(self):
+        yield _get_fake_redis_client().pipeline()
+
+    def basic_publish(self, message, exchange='', routing_key='', **kwargs):
+        self._inplace_augment_message(message, exchange, routing_key)
+        # anon exchange: routing_key is the destination queue
+        return self._put(routing_key, message, **kwargs)
+
+
+class FakeRedisKombuTransportLite(kombu_redis.Transport):
+    Channel = FakeRedisKombuChannelLite
+
+    def __init__(self, *args, **kwargs):
+        super(FakeRedisKombuTransportLite, self).__init__(*args, **kwargs)
 
     def _get_errors(self):
         return ((KeyError,), (IndexError,))
 
 
+class FakeRedisKombuTransport(FakeRedisKombuTransportLite):
+    Channel = FakeRedisKombuChannel
+
+
 @skip.unless_module('redis')
-class test_Channel:
+class TestRedisChannel(unittest.TestCase):
```

Comment:
can't we use pytest alternative here?

Reply:
I mean probably, although I didn't understand why kombu includes its own version that it already uses everywhere, and I didn't want to introduce too much scope creep for this already overly large set of changes/code diff.

Reply:
@auvipy I am proposing it is a good suggestion that we open as a new ticket, after the next kombu release (or you can open the ticket now).  However I want to suggest something that I think could be awesome for this project and the other Celery dependencies, but lets start with Kombu.  Following the next release of Kombu I would like us to find like 2 determine and schedule two time slots a week for up to 1-hour each that are recurring optional invites and when available we join and groom the Issues report.  By treating that backlog as an Agile scrum backlog of sorts we can clean it up as a group, I mean if myself or you or someone else knows of issues that were fixed and we can relate that back to open issues in a group session and determine which tickets to close and update we can get the list much much smaller than in a shorter amount of time than working entirely disparately.   

Also, for me I became an Agile CSM -- Certified SCRUM Master in October of 2015 and have been serving as a Developer and CSM for my team at work since that time.  They recently changed requirements for renewal of the certification to do additional continuing education units, and I am hoping to petition them that opens source work such as this should be able to count for some of those hours when paying the renewal fees.  Let me know what you think, we can get some kind of WexEx link that we can join and crush this backlog in prep for Celery 5!